### PR TITLE
Build ProjectTemplates in Source-Build

### LINKED ISSUE #40637: Build template packages in servicing
Currently building release/6.0 doesn't produce the template packages.

Microsoft.DotNet.Web.ItemTemplates
Microsoft.DotNet.Web.ProjectTemplates
Microsoft.DotNet.Web.Spa.ProjectTemplates

This doesn't work well for .NET source-build which needs to product the entire product for every build (e.g. previews, rtm, servicing, etc).

Would it be amenable to always produce the template packages during servicing builds?  An alternative less desirable option would be to build them only in source-build.

Currently source-build treats these packages as text-only packages that are checked in/reverse engineered. This solution is not ideal as it requires manual updates, is a maintenance burden, and in general is problematic.

This issue is similar in nature to https://github.com/dotnet/aspnetcore/issues/39471 and https://github.com/dotnet/templating/issues/4337.

### PR DESCRIPTION
**Pull request by wtgodbe**
Fixes https://github.com/dotnet/aspnetcore/issues/40637

Test build: https://dev.azure.com/dnceng/internal/_build/results?buildId=1656862&view=results

Will backport to 6.0 once the test build looks good.

### CODE REVIEW

File: Directory.Build.targets [hunk 1]
```diff
@@ -3,7 +3,12 @@
     <!-- Only build Microsoft.AspNetCore.App, Microsoft.AspNetCore.App.Ref, and ref/ assemblies in source build. -->
```
Comment (MichaelSimons) [inline Directory.Build.targets:3]: This comment is now out of date.

**Review by MichaelSimons — APPROVED** [review]

**Comment by wtgodbe** [comment]
Missed the templates from the submodule - new build: https://dev.azure.com/dnceng/internal/_build/results?buildId=1657168&view=results

File: Directory.Build.targets (hunk 1 above)
Comment (wtgodbe) [reply Directory.Build.targets]: Fixed

**Pull request merged**